### ISSUE TITLE
Error when assuming a role with a path

### ISSUE THREAD

**Issue by paolofulgoni** [Noovolari/leapp#45]
**Describe the bug**

I configured a "truster" account. When I try to activate it I get an error.
The configured role contains a [path](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-friendly-names), e.g. `pathpart1/pathpart2/rolename`

**Leapp Version**

0.3.2

**To Reproduce**
1. configure a "truster" account where the role has a path
2. activate the account

**Desktop (please complete the following information):**
 - OS: Windows
 - OS Version 10
 - Leapp Version 0.3.2

**Additional context**

This is the error I see in the log file (I replaced the actual role name):

```
[2020-11-19 22:54:51.923] [error] [renderer] [t] Error in assume role from plain to truster in get session token:  ValidationError: 1 validation error detected: Value 'truster-on-xxx/yyy/rolename' at 'roleSessionName' failed to satisfy constraint: Member must satisfy regular expression pattern: [\w+=,.@-]*
    at constructor.extractError (file:///C:/Users/fulgoni/AppData/Local/Programs/Leapp/resources/app.asar/dist/leapp-client/main.120160f17d385276afa2.js:2:3499698)
    at constructor.callListeners (file:///C:/Users/fulgoni/AppData/Local/Programs/Leapp/resources/app.asar/dist/leapp-client/main.120160f17d385276afa2.js:2:2539143)
    at constructor.emit (file:///C:/Users/fulgoni/AppData/Local/Programs/Leapp/resources/app.asar/dist/leapp-client/main.120160f17d385276afa2.js:2:2538853)
    at constructor.emitEvent (file:///C:/Users/fulgoni/AppData/Local/Programs/Leapp/resources/app.asar/dist/leapp-client/main.120160f17d385276afa2.js:2:2723145)
    at constructor.e (file:///C:/Users/fulgoni/AppData/Local/Programs/Leapp/resources/app.asar/dist/leapp-client/main.120160f17d385276afa2.js:2:2718652)
    at r.runTo (file:///C:/Users/fulgoni/AppData/Local/Programs/Leapp/resources/app.asar/dist/leapp-client/main.120160f17d385276afa2.js:2:753905)
    at file:///C:/Users/fulgoni/AppData/Local/Programs/Leapp/resources/app.asar/dist/leapp-client/main.120160f17d385276afa2.js:2:754111
    at constructor.<anonymous> (file:///C:/Users/fulgoni/AppData/Local/Programs/Leapp/resources/app.asar/dist/leapp-client/main.120160f17d385276afa2.js:2:2718922)
    at constructor.<anonymous> (file:///C:/Users/fulgoni/AppData/Local/Programs/Leapp/resources/app.asar/dist/leapp-client/main.120160f17d385276afa2.js:2:2723201)
    at constructor.callListeners (file:///C:/Users/fulgoni/AppData/Local/Programs/Leapp/resources/app.asar/dist/leapp-client/main.120160f17d385276afa2.js:2:2539249)
```

It seems the role name is concatenated to `truster-on-` for the session name, but the `/` char is not supported. If so, removing all the chars which don't match `[\w+=,.@-]` may fix the issue.

**Comment by ericvilla** [comment]
Thanks @paolofulgoni! The regex that we applied in the validation phase derives from the one applied by AWS when you try to create a role. I will attach a screenshot that better explains what I mean :)

<img width="892" alt="Screenshot 2020-11-19 at 23 57 42" src="https://user-images.githubusercontent.com/15871492/99734438-18eb5f80-2ac3-11eb-96e2-4b7429f0b671.png">

**Comment by paolofulgoni** [comment]
Hi @ericvilla thanks for your reply!

The path is not in the role name, but in a separate field:
![image](https://user-images.githubusercontent.com/5703017/99775597-d734d600-2b0f-11eb-9ff8-15851fcee1c5.png)

When I configured the "truster" account in Leapp, I wrote the path+rolename, because the path is part of the role ARN. Is that correct?

**Comment by urz9999** [comment]
Ok I got it! Probably we can also fix this as part of a next release, it's just another way to construct role ARN if the path parameter is added to the role name

**Comment by ericvilla** [comment]
@paolofulgoni I confirm that we need to cover this scenario too. That was an important bug report! Thank you.

**Comment by paolofulgoni** [comment]
I suspect the issue is due to these two lines:
https://github.com/Noovolari/leapp/blob/2ce5f48c3c2632f79126d2b2cef3627e7bd028d5/src/app/strategies/awsStrategy.ts#L216
https://github.com/Noovolari/leapp/blob/2ce5f48c3c2632f79126d2b2cef3627e7bd028d5/src/app/services/workspace.service.ts#L379

A simple solution could be to replace all the chars in the role name which are not allowed in the session name (i.e. `[^\w+=,.@-]`) with a dash... I don't know TS but it could be something like this `rolename.replace(/[^\w+=,.@-]/, "-");` .
I think the session name is not that important, do you agree?

**Comment by andreacavagna01** [comment]
A string replace could be a great idea @paolofulgoni, I will look at it late this week!

**Comment by ericvilla** [comment]
Hi @paolofulgoni, I think we could solve this during account creation. As Andrea said, I think we're gonna investigate it further late this week, but I will share with you the service in which accounts are actually created: provider-manager.service.ts

**Comment by paolofulgoni** [comment]
Ok, thanks!
I see from the AWS docs that the max length of RoleSessionName is 64 chars. This could be an additional check to do before calling the API (I haven't checked if it's done already)
https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html#API_AssumeRole_RequestParameters

**Comment by ericvilla** [comment]
Hi @paolofulgoni, we solved this bug in the new [v0.4.1 release](https://github.com/Noovolari/leapp/projects/5). Now the region option is specified when invoking the `aws ssm start-session` command

**Comment by paolofulgoni** [comment]
Hi @ericvilla, sorry for replying late.
I confirm the bug is solved, thank you!!!! 🎉
